src/sentry/issues/endpoints/group_open_periods.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+from typing import TYPE_CHECKING, Any
+
+from django.utils import timezone
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import EnvironmentMixin, region_silo_endpoint
+from sentry.api.bases import GroupEndpoint
+from sentry.api.paginator import GenericOffsetPaginator
+from sentry.incidents.utils.metric_issue_poc import OpenPeriod
+from sentry.models.activity import Activity
+from sentry.types.activity import ActivityType
+
+if TYPE_CHECKING:
+    from sentry.models.group import Group
+
+
+def get_open_periods_for_group(
+    group: Group,
+    offset: int | None = None,
+    limit: int | None = None,
+) -> list[OpenPeriod]:
+    if not features.has("organizations:issue-open-periods", group.organization):
+        return []
+
+    activities = Activity.objects.filter(
+        group=group,
+        type__in=[ActivityType.SET_UNRESOLVED.value, ActivityType.SET_RESOLVED.value],
+        datetime__gte=timezone.now() - timedelta(days=90),
```

Comment:
do we really need to pull all 90 days of open periods every request? shouldn't we be able to look at the `statsPeriod` to minimize this?

Reply:
also: can you use offset/limit here to limit the amount of results you pull from the database? if you have a limit of 10, will that equate to 20 activity rows?